src/aggregator/server/rawtcp/server.go

```diff
@@ -196,48 +196,52 @@ func (s *handler) Handle(conn net.Conn) {
 				zap.String("remoteAddress", remoteAddress),
 				zap.Error(err),
 			)
-		case addUntimedError:
-			s.metrics.addUntimedErrors.Inc(1)
-			s.log.Error("error adding untimed metric",
-				zap.String("remoteAddress", remoteAddress),
-				zap.Stringer("type", untimedMetric.Type),
-				zap.Stringer("id", untimedMetric.ID),
-				zap.Any("metadatas", stagedMetadatas),
-				zap.Error(err),
-			)
-		case addForwardedError:
-			s.metrics.addForwardedErrors.Inc(1)
-			s.log.Error("error adding forwarded metric",
-				zap.String("remoteAddress", remoteAddress),
-				zap.Stringer("id", forwardedMetric.ID),
-				zap.Time("timestamp", time.Unix(0, forwardedMetric.TimeNanos)),
-				zap.Float64s("values", forwardedMetric.Values),
-				zap.Error(err),
-			)
-		case addTimedError:
-			s.metrics.addTimedErrors.Inc(1)
-			s.log.Error("error adding timed metric",
-				zap.String("remoteAddress", remoteAddress),
-				zap.Stringer("id", timedMetric.ID),
-				zap.Time("timestamp", time.Unix(0, timedMetric.TimeNanos)),
-				zap.Float64("value", timedMetric.Value),
-				zap.Error(err),
-			)
-		case addPassthroughError:
-			s.metrics.addPassthroughErrors.Inc(1)
-			s.log.Error("error adding passthrough metric",
-				zap.String("remoteAddress", remoteAddress),
-				zap.Stringer("id", timedMetric.ID),
-				zap.Time("timestamp", time.Unix(0, timedMetric.TimeNanos)),
-				zap.Float64("value", timedMetric.Value),
-				zap.Error(err),
-			)
 		default:
-			s.metrics.unknownErrorTypeErrors.Inc(1)
-			s.log.Error("unknown error type",
-				zap.String("errorType", fmt.Sprintf("%T", err)),
-				zap.Error(err),
-			)
+			switch current.Type {
+			case encoding.CounterWithMetadatasType:
+				fallthrough
```

Comment:
Huh, TIL `fallthrough`

Reply:
a lot more sane than the implicit fall through without a break.

remember that ios cert vulnerability back in the day?